staging/src/k8s.io/apiserver/pkg/util/openapi/enablement.go

```diff
@@ -45,14 +46,57 @@ func restoreDefinitions(defs map[string]common.OpenAPIDefinition) {
 	if !utilfeature.DefaultFeatureGate.Enabled(genericfeatures.OpenAPIEnums) {
 		for gvk, def := range defs {
 			orig := &def.Schema
-			if ret := pruneEnums(orig); ret != orig {
+
+			ret := pruneEnums(orig)
+			ret = pruneUnrecognizedExtensions(ret)
+
+			if ret != orig {
 				def.Schema = *ret
 				defs[gvk] = def
 			}
 		}
 	}
 }
 
+var whitelistedExtensions sets.String = func() sets.String {
+	res := sets.NewString(
+		"x-kubernetes-list-type",
+		"x-kubernetes-map-keys",
+		"x-kubernetes-map-type",
+		"x-kubernetes-patch-strategy",
+		"x-kubernetes-patch-merge-key",
+	)
+
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.DeclarativeValidationsInOpenAPI) {
+		res.Insert("x-kubernetes-validations")
+	}
+
+	return res
+}()
+
+func pruneUnrecognizedExtensions(schema *spec.Schema) *spec.Schema {
```

Comment:
"unrecognizedextensions" doesn't really make sense, we don't really allow arbitrary extensions anyway.

Reply:
Still unclear which extension you're trying to remove here?

Reply:
Just trying not to expose any new extensions that are in the OpenAPI unless the feature is enabled.

Reply:
I'm still not sure why this is needed all of a sudden, not super happy with the allow list, can we just remove all the value validation from built-ins only?